test/functional/test_framework/messages.py

```diff
@@ -1045,13 +1045,13 @@ def deserialize(self, f):
 
         self.nStartingHeight = struct.unpack("<i", f.read(4))[0]
 
-        if self.nVersion >= 70001:
-            # Relay field is optional for version 70001 onwards
-            try:
-                self.relay = struct.unpack("<b", f.read(1))[0]
-            except:
-                self.relay = 0
-        else:
+        # Relay field is optional for version 70001 onwards
+        # But, unconditionally check it to match behaviour in bitcoind
+        try:
+            self.relay = struct.unpack("<b", f.read(1))[0]
+        except KeyboardInterrupt:
+            raise KeyboardInterrupt
+        except:
```

Comment:
I think better would be to just catch `struct.error`:

```suggestion
        except struct.error:
            self.relay = 0
```

and allow all other exceptions to raise.

Reply:
addressed in 5c46992

Reply:
You can remove:

```
        except KeyboardInterrupt:
            raise KeyboardInterrupt
```

since there's no blanket `except` catching, this is doing nothing (`KeyboardInterrupt` exceptions would be raised anyway)